src/api/app/models/bs_request.rb

```diff
@@ -107,6 +108,21 @@ def check_creator
     errors.add(:creator, "Login #{user.login} is not an active user")
   end
 
+  def double_actions
+    bs_request_actions.each do |actionA|
+      bs_request_actions.each do |actionB|
+        next if actionA == actionB
+        next if actionA.type == "add_role" # the exception that proves the rule
+        next if actionA.type == "maintenance_incident" # we could extend the check for package link target
+        if actionA.type == actionB.type \
+           && actionA.target_project == actionB.target_project \
+           && actionA.target_package == actionB.target_package
+          errors.add(:invalid, "double action for #{actionB.try(:target_project)}/#{actionB.try(:target_package)} ")
+        end
+      end
+    end
```

Comment:
This looks strange and inefficient. First of all you could filter out types that are going to be skipped on db level (where).
Second looping twice over the same dataset is just wrong. And since this is a validation, why can't we just query for existence of a matching object?